core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2350,6 +2350,16 @@ object SparkContext extends Logging {
     }
   }
 
+  private[spark] def getActiveContext(): Option[SparkContext] = {
+    SPARK_CONTEXT_CONSTRUCTOR_LOCK.synchronized {
+      Option(activeContext.get())
+    }
+  }
+
+  private[spark] def stopActiveContext(): Unit = {
```

Comment:
Shouldn't this be unnecessary with more carefully written tests? that always close the context etc when done?